.bit/tests/test.1.5.js

```diff
@@ -21,26 +21,48 @@ if (uri[0] != "h") {
    throw new Error("You have not added your function url as a secret!");
 }
 
+//If we have no query string then add one
+//this allows us to append without error
+if(uri.indexOf("?") === -1)
+{
+    uri = uri + "?x=1"
+}
+
 try {
     (async () => {
-        const resp = await fetch(uri + "&password=letmein", {
+
+        const uriWithQuery  = uri + "&password=letmein"
+
+        const resp = await fetch(uriWithQuery, {
             method: 'GET'
         });
+
+        if(resp.status == 404){
+            console.error(`Your function could not be found at "${uriWithQuery}" check function url secret 🔍`);
+            process.exit(1)
+        }
+
+        if(resp.status == 500){
+            console.error("Your function has an error and could not be run 🐛");
+            process.exit(1)
+        }
+
         var correct = await resp.text()
 
         const response = await fetch(uri + "&password=incorrect", {
             method: 'GET'
         });
+
         var incorrect = await response.text()
 
         try {
             if (correct == "Access granted." && incorrect == "Access denied.") {
                 console.log("Yay! 🎉 You didn't let the bad guys in.")
             } else {
                 console.log("Try again!")
-                console.log(`We submitted "letmein" and got ${correct}, which should equal "Access granted."`)
-                console.log(`We submitted "incorrect" and got ${incorrect}, which should equal "Access granted."`)
-                console.log("Make sure your function has an authorization level of 'Function.'")
+                console.log(`We submitted "letmein" and got "${correct}", which should equal "Access granted."`)
+                console.log(`We submitted "incorrect" and got "${incorrect}", which should equal "Access denied."`)
+                console.log("Make sure your function has an authorization level of 'anonymous'")
```

Comment:
Since you have added the conditionals that append a querystring if we don't have one, then this suggestion regarding authorisation level is not needed.
> Previously, I added the "make sure your function has an authorization level of Function" because with this level, the function URL comes with a querystring. [function URL]?code=ladkfjalkfjlsfjalj Reference: https://vincentlauzon.com/2017/12/04/azure-functions-http-authorization-levels/. Since Anonymous does not require authentication, it does not have the ?code=adlfkja parameter and therefore lacks a querystring.